modules/lib/src/main/resources/assets/js/app/create/NewContentDialog.ts

```diff
@@ -301,6 +304,48 @@ export class NewContentDialog
             .finally(() => this.handleTypesLoaded());
     }
 
+    private updateUploaderState(): void {
+        this.newContentUploader.reset();
+
+        if (!this.parentContent) {
+            this.handleUploaderAllowed();
+        } else if (this.isTemplateFolderSelected()) {
+            this.handleUploaderNotAllowed();
+        } else {
+            new GetContentTypeByNameRequest(this.parentContent.getType()).sendAndParse().then((type: ContentType) => {
+                this.handleUploaderByContentType(type);
+            }).catch(DefaultErrorHandler.handle);
+        }
+    }
+
+    private handleUploaderAllowed(): void {
+        this.newContentUploader.setVisible(true);
+        this.newContentUploader.setEnabled(true);
+        this.removeClass('no-uploader-el');
+    }
+
+    private handleUploaderNotAllowed(): void {
+        this.addClass('no-uploader-el');
+        this.newContentUploader.setEnabled(false);
+        this.newContentUploader.setVisible(false);
+    }
+
+    private handleUploaderByContentType(parentContentType: ContentType): void {
+        if (parentContentType.getAllowChildContentType()?.length > 0) {
+            if (parentContentType.getAllowChildContentType().some((allowedType: string) => this.isMediaType(allowedType))) {
```

Comment:
This logic belongs in the `ContentTypeSummary` class. We also have static `ContentTypeName.MEDIA` and `ContentTypeName.isDescendantOfMedia` which should be used.

Here it should be something very simple, like

```Typescript
private updateUploaderState(): void {
  this.newContentUploader.reset();
  
  if (!this.parentContent) {
      this.setUploaderAllowed(true);
  } else if (this.isTemplateFolderSelected()) {
      this.setUploaderAllowed(false);
  } else {
      new GetContentTypeByNameRequest(this.parentContent.getType())
  .sendAndParse()
  .then((contentType: ContentType) => this.setUploaderAllowed(contentType.isMediaChildContentAllowed()))
  .catch(DefaultErrorHandler.handle);
  }
}
```